backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -995,6 +1027,71 @@ public void legacyRbacUpdateEventTypesLinkedToEndpoint(final SecurityContext sec
+                behaviorGroupRepository.delete(orgId, behaviorGroup.getId());
+            } else {
+                behaviorGroupRepository.deleteEndpointFromBehaviorGroup(behaviorGroup.getId(), endpointId, orgId);
+            }
+        }
+
+        // Create or update relevant behavior groups
+        createOrUpdateLinkedBehaviorGroup(eventTypeIds, endpointId, updatedEndpoint.getName(), orgId);
+    }
+
+    private void createOrUpdateLinkedBehaviorGroup(Set<UUID> eventTypeIds, UUID endpointId, String endpointName, String orgId) {
+        String behaviorGroupName = String.format("Integration \"%s\" behaviour group", endpointName);
+
+        // group event types by bundle
+        Map<UUID, Set<UUID>> eventTypesGroupedByBundle = new HashMap<>();
+        for (UUID eventTypeId : eventTypeIds) {
+            Optional<Bundle> bundle = eventTypeRepository.findBundleByEventTypeId(eventTypeId);
+            if (bundle.isPresent()) {
+                if (eventTypesGroupedByBundle.containsKey(bundle.get().getId())) {
+                    eventTypesGroupedByBundle.get(bundle.get().getId()).add(eventTypeId);
+                } else {
+                    Set<UUID> eventTypes = new HashSet<>();
+                    eventTypes.add(eventTypeId);
+                    eventTypesGroupedByBundle.put(bundle.get().getId(), eventTypes);
+                }
+            }
+        }
+
+        for (UUID bundleId : eventTypesGroupedByBundle.keySet()) {
+            Optional<BehaviorGroup> existingBg = behaviorGroupRepository.findBehaviorGroupsByName(orgId, bundleId, behaviorGroupName);
+            if (existingBg.isPresent()) {
+                Boolean alreadyAssociatedAction = existingBg.get().getActions().stream().anyMatch(bga -> bga.getId().endpointId.equals(endpointId));
+
+                if (!alreadyAssociatedAction) {
+                    int position = existingBg.get().getActions().stream().mapToInt(ba -> ba.getPosition()).max().orElse(-1) + 1;
+                    behaviorGroupRepository.appendActionToBehaviorGroup(orgId, existingBg.get().getId(), endpointId, position);
+                }
+                for (UUID eventTypeId : eventTypesGroupedByBundle.get(bundleId)) {
+                    Boolean alreadyAssociatedEventType = existingBg.get().getBehaviors().stream().anyMatch(bh -> bh.getId().eventTypeId.equals(eventTypeId));
+                    if (!alreadyAssociatedEventType) {
+                        behaviorGroupRepository.appendBehaviorGroupToEventType(orgId, existingBg.get().getId(), eventTypeId);
+                    }
+                }
+            } else {
+                // Create or update legacy behavior group structure
+                BehaviorGroup behaviorGroup = new BehaviorGroup();
+                behaviorGroup.setBundleId(bundleId);
+                behaviorGroup.setDisplayName(behaviorGroupName);
+
+                behaviorGroupRepository.createFull(
+                    null,
```

Comment:
I think we should provide the account ID value here until we decide to remove it from the DB table for everyone.